### PR TITLE
web: add backup & delete account on settings

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are adding section to `back up` or `delete account` in the `Settings` page.

### CODE REVIEW

File: web-frontend/src/custom/api/useDelete.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  var url = `${import.meta.env.VITE_BASE_API}/${resource}/${nickname}`;
+
+  return useMutation<T, unknown, T, unknown>({
+    mutationKey: [resource],
+    mutationFn: async () => {
+      const { data } = await axios.delete<T>(url);
+      return data;
+    },
+  });
+}
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply web-frontend/src/custom/api/useDelete.ts]: Done

**Review by mazmassa — APPROVED** [review]

File: web-frontend/src/custom/api/useDelete.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply web-frontend/src/custom/api/useDelete.ts]: Example of `useDelete` request :

```typescript
// STYLES

// EXTERNALS
import axios from 'axios';
import { UseMutationResult, useMutation } from '@tanstack/react-query';

// LOCALS

function useDelete<T>(
  resource: string,
): UseMutationResult<T, unknown, T, unknown> {
  var url = `${import.meta.env.VITE_BASE_API}/${resource}`;

  return useMutation<T, unknown, T, unknown>({
    mutationKey: [resource],
    mutationFn: async () => {
      const { data } = await axios.delete<T>(url);

      return data;
    },
  });
}

export default useDelete;

```

File: web-frontend/src/layouts/WalletLayout/WalletLayout.tsx [hunk 2]
```diff
@@ -114,6 +114,7 @@ function WalletLayout(props: WalletProps) {
   }));
 
   const selectedAccountKey: number = parseInt(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (Thykof) [inline web-frontend/src/layouts/WalletLayout/WalletLayout.tsx:117]: haha

**Review by Thykof — APPROVED** [review]

File: web-frontend/src/layouts/WalletLayout/WalletLayout.tsx (hunk 2 above)
Comment (mazmassa) [reply web-frontend/src/layouts/WalletLayout/WalletLayout.tsx:117]: Smart people man! 🤣

**Pull request merged**